chpl/chpl-service/src/main/java/gov/healthit/chpl/web/controller/UserManagementController.java

```diff
@@ -335,34 +421,59 @@ public String grantUserRole(@RequestBody GrantRoleJSONObject grantRoleObj) throw
         } else {
             userManager.grantRole(user.getSubjectName(), grantRoleObj.getRole());
         }
-
+        
         UserDTO updated = userManager.getByName(grantRoleObj.getSubjectName());
-
+        
         String activityDescription = "User " + user.getSubjectName() + " was granted role " + grantRoleObj.getRole()
                 + ".";
         activityManager.addActivity(ActivityConcept.ACTIVITY_CONCEPT_USER, user.getId(), activityDescription, user,
                 updated);
-
+        
         return "{\"roleAdded\" : true}";
-
     }
 
-    @ApiOperation(value = "Remove roles previously granted to a user.",
+    
+    @Deprecated
+    @ApiOperation(value = "DEPRECATED.  Remove roles previously granted to a user.",
             notes = "Users may be given ROLE_ADMIN, ROLE_ACB, "
                     + "ROLE_ATL, or ROLE_ONC_STAFF roles within the system.")
     @RequestMapping(value = "/revoke_role", method = RequestMethod.POST, consumes = MediaType.APPLICATION_JSON_VALUE,
             produces = "application/json; charset=utf-8")
-    public String revokeUserRole(@RequestBody GrantRoleJSONObject grantRoleObj) throws InvalidArgumentsException,
+    public String revokeUserRoleDeprecated(@RequestBody GrantRoleJSONObject grantRoleObj) throws InvalidArgumentsException,
             UserRetrievalException, UserManagementException, UserPermissionRetrievalException, JsonProcessingException,
             EntityCreationException, EntityRetrievalException {
 
+        return revoke(grantRoleObj);
+    }
+    
+    @ApiOperation(value = "Remove roles previously granted to a user.",
+            notes = "Users may be given ROLE_ADMIN, ROLE_ACB, "
+                    + "ROLE_ATL, or ROLE_ONC_STAFF roles within the system.")
+    @RequestMapping(value = "/{userName}/roles/{roleName}", method = RequestMethod.DELETE, 
```

Comment:
Would think this is a PUT also, as it's not deleting the user, but it is editing it

Reply:
This deletes the relationship between a user and role, therefore  I think it should be a delete.

Reply:
Plus, if this is a PUT and the one above is a PUT, it would be ambiguous.